src/hangar/dataset.py

```diff
@@ -429,6 +421,26 @@ def _backend(self) -> str:
         '''
         return self._dflt_backend
 
+    def _cm_setup(self):
```

Comment:
I'm very confused here. Why are we moving the methods to handle context manager setup and teardown out of the standard python methods? 

Am I missing something? Or are we just redirecting the call? If so. Consider that this breaks the manual attributes set on the weakref object proxy which is returned to the user upon access. (a `functools partial` instance applied with all arguments filled but unevaluated to call the CM protocol of a weakref proxy via manually set attributes on a reference python doesn't normally think is `hashable`)  

At best you might describe it as an elegant hack which is thoroughly tested. At worst it's a mindbend 2 lines of code applied librally to the User facing classes which took me a week to write, will give most python experts a significant pause when they see it, and will just be ignored by everyone else. 

I don't want to be a pain and make others pay for a difficult design decision, but if we don't have a strong need to break with standard python CM conventions, it'll be easier to use them all around.

Reply:
So none of this is well tested, I have just made it up (which I should not have I have verified it well I guess). So the reason for taking it out of standard methods is, I can rely on `self._is_conman` while re-enabling it after the commit. Otherwise, my `conman` becomes False on `__exit__` and I'll have to keep a reference separately. I mean, that's OK and probably there could be other methods but I just found this easy. Thanks for the pointer, I'll check it out.